src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+           auto bank_grp_idx_1 = kernel.group_id(1);
+    
+           auto input_buffer = xrt::bo(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, bank_grp_idx_0);
+           auto output_buffer = xrt::bo(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, bank_grp_idx_1);
+
+In the above code ``xrt::bo`` buffer objects are created using the class's constructor. All the arguments are identical to the ``xrtBOAlloc`` API as discussed above in the C example explanation.  
+
+
+2. Data transfer using Buffers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+XRT Buffer API library provides a rich set of APIs helping the data transfers between the host and the device, between the buffers, etc. We will discuss the following data transfer style
+
+        I. Data transfer between host and device by Buffer read/write API
+        II. Data transfer between host and device by Buffer map API
+        III. Data transfer between buffers by copy API
+
+
+I. Data transfer between host and device by Buffer read/write API
+*****************************************************************
+
+To transfer the data from the host to the device, the user first needs to update the host-side buffer backing pointer followed by a DMA transfer to the device. 
+
+The following C APIs are used for the above tasks
+
+    1. ``xrtBOWrite``  
+    2. ``xrtBOSync`` with flag ``XCL_BO_SYNC_BO_TO_DEVICE``
+    
+In C++, ``xrt::bo`` class has following methods for the same functionality
+
+    1. ``xrt::bo::write``
+    2. ``xrt::bo::sync`` with flag ``XCL_BO_SYNC_BO_TO_DEVICE``
+
+To transfer the data from the device to the host, the steps are reverse, the user first needs to do a DMA transfer from the device followed by the reading data from the host-side buffer backing pointer. 
+
+The following C APIs are used for the above tasks
+
+     1. ``xrtBOSync`` with flag ``XCL_BO_SYNC_BO_FROM_DEVICE``
+     2. ``xrtBORead``
+
+In C++ the corresponding ``xrt::bo`` class's methods are
+
+    1. ``xrt::bo::sync`` with flag ``XCL_BO_SYNC_BO_FROM_DEVICE``
+    2. ``xrt::bo::read``
+
+
+Code example of transferring data from the host to the device
+
+.. code:: c
+      :number-lines: 20
+           
+           xrtBufferHandle input_buffer = xrtBOAlloc(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, bank_grp_idx_0);
+
+           // Prepare the input data
+           int buff_data[data_size];
+           for (int i=0; i<data_size; ++i) {
+               buff_data[i] = i;
+           }
+    
+           xrtBOWrite(input_buffer,buff_data,data_size*sizeof(int),0);
+           xrtSyncBO(input_buffer,XCL_BO_SYNC_BO_TO_DEVICE, data_size*sizeof(int),0);
+    
+
+**C++**: The equivalent C++ API based code
+
+
+.. code:: c++
+      :number-lines: 20    
+           
+           auto input_buffer = xrt::bo(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, bank_grp_idx_0);
+           // Prepare the input data
+           int buff_data[data_size];
+           for (auto i=0; i<data_size; ++i) {
```

Comment:
That's fine, the heading of this is C++ API based code, not C++ code. The idea is to have the identical code only API change so that the reader understands quickly